File: examples/tcp_example.php
<?php

require dirname(__FILE__) . '/../drivers/php/hurricane.php';

date_default_timezone_set('America/Denver');

$gateway = new \Hurricane\Gateway(
    new \Erlang\SocketWrapper('localhost', '3000')
);
while (true) {
    $request = \Hurricane\Message::create()
        ->setType('request')
        ->setDestination('time_server')
        ->setTag(0)
        ->setData(null);
    $gateway->send($request);
    $response = $gateway->recv();
    echo $response->getData() . PHP_EOL;
}

File: examples/TcpExample.java
import org.hurricane.Gateway;
import org.hurricane.Message;

public class TcpExample {
    public static void main(String[] args) throws Exception {
        Gateway gateway = new Gateway("localhost", 3000);

        while (true) {
            Message request = new Message();
            request.setType("request");
            request.setDestination("time_server");
            request.setTag(0);
            request.setData("");

            System.out.println(request);
            gateway.send(request);
            System.out.println(gateway.recv());
        }
    }
}
